x-pack/plugins/monitoring/init.js

```diff
@@ -21,25 +28,61 @@ import { createCollectorSet } from './server/kibana_monitoring';
  * @param server {Object} HapiJS server instance
  */
 export const init = (monitoringPlugin, server) => {
-  monitoringPlugin.status.yellow('Initializing');
-  const xpackMainPlugin = server.plugins.xpack_main;
+  const config = server.config();
 
-  xpackMainPlugin.status.once('green', async () => {
-    const config = server.config();
+  const collectorSet = new CollectorSet(server);
+  server.expose('collectorSet', collectorSet); // expose the collectorSet service
+  /*
+   * Register collector objects for stats to show up in the APIs
+   */
+  collectorSet.register(getOpsStatsCollector(server));
+  collectorSet.register(getKibanaUsageCollector(server));
+  collectorSet.register(getSettingsCollector(server));
```

Comment:
Registering a collector doesn't need to be delayed on anything. It's only when fetch is called on them that we want to try to make sure Elasticsearch is up. The `bulkUploader` handles this by watching elasticsearch plugin status. This isn't happening robustly from the API caller yet, but I have some plans to make that better in a later PR.